### PR TITLE
RestHighLevelClient to not implement Closeable

### PR DESCRIPTION
**Pull request by javanna**
The `RestHighLevelClient` should accept an instance of the low-level client externally created, and managed. The high level client shouldn't create nor close the underlying low level client, this way it also doesn't need to implement `Closeable`.

### CODE REVIEW

**Comment by javanna** [comment]
@tlrx @cbuescher can you have a look please?

File: client/highlevel-rest/src/test/java/org/elasticsearch/client/HLClientSearchIT.java [hunk 1]
```diff
@@ -45,11 +44,11 @@
 
 public class HLClientSearchIT extends ESRestTestCase {
 
-    private HighlevelClient aClient;
+    private RestHighLevelClient highLevelClient;
```
Comment (tlrx) [inline client/highlevel-rest/src/test/java/org/elasticsearch/client/HLClientSearchIT.java]: Let's call it `client`?

**Review by tlrx — APPROVED** [review]
Left a minor comment - LGTM

**Comment by cbuescher** [comment]
LGTM

File: client/highlevel-rest/src/test/java/org/elasticsearch/client/HLClientSearchIT.java (hunk 1 above)
Comment (javanna) [inline client/highlevel-rest/src/test/java/org/elasticsearch/client/HLClientSearchIT.java]: this test is going to be removed together with the search stuff that we did till now, so I am not going to bother renaming the instance member, sorry :)

**Pull request merged**